apps/server/src/trpc/list.ts

```diff
@@ -1,39 +1,79 @@
-import { flatten, validModuleRegex } from '@modtree/utils'
+import { flatten } from '@modtree/utils'
 import { z } from 'zod'
 import { createRouter } from './router'
 import { api } from '../main'
+import { entities } from '../schemas/entities'
+import { parseCommaSeparatedString } from '../utils/parse'
 
 export const list = createRouter()
   /** list modules */
   .query('modules', {
-    input: z.array(z.string().regex(validModuleRegex)),
+    meta: {
+      openapi: {
+        enabled: true,
+        method: 'GET',
+        path: '/modules',
+      },
+    },
+    input: z.object({
+      moduleCodes: z.string().transform(parseCommaSeparatedString),
+    }),
+    output: z.array(entities.Module),
     async resolve({ input }) {
-      return api.moduleRepo.findByCodes(input)
+      return api.moduleRepo.findByCodes(input.moduleCodes)
```

Comment:
`input` properties are enforced as `ZodString`. Hence, the workaround is to pass in a comma separated string and split it into an array.

Reply:
Fixed in #384